wezterm-gui/src/termwindow/charselect.rs

```diff
@@ -268,45 +269,54 @@ fn compute_matches(selection: &str, aliases: &[Alias], group: CharSelectGroup) -
-                        if codepoints == *sel {
-                            Some(MatchResult {
+        for (row_idx, entry) in aliases.iter().enumerate() {
+            let glyph = entry.glyph();
+            let alias_result = matcher
+                .fuzzy_match(&entry.name, selection)
+                .map(|score| MatchResult::new(row_idx, score, selection, aliases));
+            if let Some(value) = match &numeric_selection {
+                Some(sel) => {
+                    let codepoints = entry.codepoints();
+                    if codepoints == *sel {
+                        Some(MatchResult {
+                            row_idx,
+                            score: i64::max_value(),
+                        })
+                    } else {
+                        let number_result = matcher
+                            .fuzzy_match(&codepoints, &sel)
+                            .map(|score| MatchResult::new(row_idx, score, sel, aliases));
+
+                        match (alias_result, number_result) {
+                            (
+                                Some(MatchResult { score: a, .. }),
+                                Some(MatchResult { score: b, .. }),
+                            ) => Some(MatchResult {
                                 row_idx,
-                                score: i64::max_value(),
-                            })
-                        } else {
-                            let number_result = matcher
-                                .fuzzy_match(&codepoints, &sel)
-                                .map(|score| MatchResult::new(row_idx, score, sel, aliases));
-
-                            match (alias_result, number_result) {
-                                (
-                                    Some(MatchResult { score: a, .. }),
-                                    Some(MatchResult { score: b, .. }),
-                                ) => Some(MatchResult {
-                                    row_idx,
-                                    score: a.max(b),
-                                }),
-                                (Some(a), None) | (None, Some(a)) => Some(a),
-                                (None, None) => None,
-                            }
+                                score: a.max(b),
+                            }),
+                            (Some(a), None) | (None, Some(a)) => Some(a),
+                            (None, None) => None,
                         }
                     }
-                    None => alias_result,
                 }
-            })
-            .collect();
+                None => alias_result,
+            } {
+                if let Some(other) = matches.get(&glyph) {
+                    // if the character was inserted already, replace it only if the score is improved
+                    if other.score < value.score {
+                        matches.insert(glyph, value);
+                    }
+                } else {
```

Comment:
How about this as a potential alternative micro optimization then?
Should avoid repeating the hash lookup for the insertion case.

```suggestion
                if let Some(other) = matches.get_mut(&glyph) {
                    // if the character was inserted already, replace it only if the score is improved
                    if other.score < value.score {
                        *other = value;
                    }
                } else {
```

Reply:
Done